### PR TITLE
Login feature test

### PR DESCRIPTION
**Pull request by Rinbo**
## Login feature test

PT: https://www.pivotaltracker.com/story/show/160590270

This test simulates a user going to the landing page and signing into the app. Factory bot is used to provide a virtual database.

### CODE REVIEW

File: features/login_to_account.feature [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  Background: 
+  Given the following user is in database
+    | email              | password  | name   |
+    | anyone@example.com | foobarbar | Anyone |  
+
+    Scenario: As a user I would like to login to my account
+    Given I am on the landing page
+    When I click on the "Login" button
+    Then I should be on the login form
```
Comment (faraznaeem) [inline features/login_to_account.feature]: You could skip this step

File: features/login_to_account.feature [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    When I fill in "Email" with "anyone@example.com"
+    And I fill in "Password" with "foobarbar"
```
Comment (faraznaeem) [inline features/login_to_account.feature]: `And I fill in "Email" with "anyone@example.com"`

File: features/login_to_account.feature [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    When I click the "Log in" button
```
Comment (faraznaeem) [inline features/login_to_account.feature]: `And I click the "Log in" button`

File: features/login_to_account.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    Then I should be on the landing page
```
Comment (faraznaeem) [inline features/login_to_account.feature]: You could skip this

File: features/login_to_account.feature [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    And I should see a message saying "Signed in successfully."
```
Comment (faraznaeem) [inline features/login_to_account.feature]: `Then I should see ...`

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -20,6 +20,7 @@ def conversation
   protected
 
   def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
     devise_parameter_sanitizer.permit(:account_update, keys: [:name])
```
Comment (faraznaeem) [inline app/controllers/application_controller.rb]: Why are you adding sanitizers for signup?

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb (hunk 6 above)
Comment (Rinbo) [reply app/controllers/application_controller.rb]: Added it to be able to signup to the app on localhost:3000 and troubleshoot

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**